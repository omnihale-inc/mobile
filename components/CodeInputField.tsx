import React from "react";
import {
  NativeSyntheticEvent,
  StyleProp,
  TextInput,
  TextInputKeyPressEventData,
  TextStyle
} from "react-native";

export type CodeInputFieldProps = {
  events: {
    onChangeTextHandler(changeValue: string): void;
    onKeyPressHandler(
      key: NativeSyntheticEvent<TextInputKeyPressEventData>
    ): void;
    onFocusHandler(): void;
    onBlurHandler(): void;
  };
  value: string;
  style: StyleProp<TextStyle>;
};

const CodeInputField = React.forwardRef(
  (
    { events, value, style }: CodeInputFieldProps,
    ref: React.ForwardedRef<TextInput | null>
  ) => {
    return (
      <TextInput
        onChangeText={events.onChangeTextHandler}
        value={value}
        style={style}
        onFocus={events.onFocusHandler}
        onBlur={events.onBlurHandler}
        keyboardType="numeric"
        ref={ref}
        onKeyPress={events.onKeyPressHandler}
      />
    );
  }
);

export default CodeInputField;
