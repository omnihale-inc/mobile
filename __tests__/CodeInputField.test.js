import renderer from "react-test-renderer";

import CodeInputField from "../components/CodeInputField";

it("render code input field", () => {
  const mockProps = {
    events: {
      onChangeTextHandler: jest.fn(),
      onKeyPressHandler: jest.fn(),
      onFocusHandler: jest.fn(),
      onBlurHandler: jest.fn()
    },
    value: "0",
    style: {},
    ref: null
  };
  const tree = renderer.create(<CodeInputField {...mockProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
