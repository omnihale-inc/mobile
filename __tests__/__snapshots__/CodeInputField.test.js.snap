// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render code input field 1`] = `
<TextInput
  keyboardType="numeric"
  onBlur={[MockFunction]}
  onChangeText={[MockFunction]}
  onFocus={[MockFunction]}
  onKeyPress={[Function]}
  style={{}}
  value="0"
/>
`;
